File: ndarray/backend/ndarray.cpp
#include <vector>
#include <string>
#include <sstream>
#include <algorithm>

#include "ndarray.h"

ndarray::ndarray(std::vector<int> &shape)
{
  shape_m = shape;
  size_m = 1;
  for (int i = 0; i < shape_m.size(); i++)
  {
    size_m *= shape_m[i];
  }
}

ndarray::~ndarray() 
{
  ;
}

std::string ndarray::to_string() const
{
  std::ostringstream oss;
  oss << "ndarray([";

  int elems = std::min(10, size_m);
  for (int i = 0; i < elems; i++)
  {
    //oss << data_m[i];
    oss << i;
    if (i < (elems - 1)) 
    { 
      oss << ", "; 
    }
  }
  oss << "], dtype=None)";

  return oss.str();
}

File: ndarray/ndarray.py
from .dtype import dtype

from ndarray_backend import ndarray as __ndarray

class ndarray(__ndarray):
    def __init__(self, shape):
        if isinstance(shape, int):
            shape = (shape,)
        super().__init__(shape)

    def __str__(self):
        return self.__repr__()

    def __repr__(self):
        return self.to_string()
